run_flax_speech_recognition_seq2seq_streaming_review.py

```diff
@@ -0,0 +1,1391 @@
+    
+    if model_args.dropout or model_args.attention_dropout or model_args.activation_dropout or model_args.encoder_dropout or model_args.decoder_dropout:
+        training_summary["hyperparameters"]["dropout"] = True
+        if model_args.dropout:
+            training_summary["hyperparameters"]["dropout_probability"] = model_args.dropout
+        if model_args.attention_dropout:
+            training_summary["hyperparameters"]["attention_dropout_probability"] = model_args.attention_dropout
+        if model_args.activation_dropout:
+            training_summary["hyperparameters"]["activation_dropout_probability"] = model_args.activation_dropout
+        if model_args.encoder_dropout:
+            training_summary["hyperparameters"]["encoder_dropout_probability"] = model_args.encoder_dropout
+        if model_args.decoder_dropout:
+            training_summary["hyperparameters"]["decoder_dropout_probability"] = model_args.decoder_dropout
+
+    # Create README if it does not exist
+    readme = output_dir / "README.md"
+    if not readme.exists():
+        readme.write_text(TrainingSummary(**training_summary).to_model_card())
+    
+    # ======================== Training ================================
+    train_start = time.time()
+
+    train_metrics = []
+    epoch = 0
+    if training_args.do_train:
+        train_dataset = vectorized_datasets["train"].shuffle(seed=training_args.seed, buffer_size=data_args.shuffle_buffer_size)
+        # Split by node
+        train_dataset = split_dataset_by_node(train_dataset, rank=current_host_idx, world_size=num_of_hosts)   
+    
+        if train_dataset.n_shards < data_args.preprocessing_num_workers:
+            num_workers = train_dataset.n_shards
+
+        logger.info(f"  Number of train dataset workers = {num_workers} {'(Capped by the number of dataset shards)' if train_dataset.n_shards < data_args.preprocessing_num_workers else ''} {'(ADVICE: In most cases you will speed up training considerably if you increase the value of --preprocessing_num_workers!)' if num_workers < 10 else ''}")
+        train_loader = data_loader(train_dataset, train_batch_size // num_of_hosts, num_workers=num_workers)
+
+    if training_args.do_eval:
+        eval_dataset = vectorized_datasets["eval"]
+    
+    if training_args.do_train and not training_args.ignore_data_skip and training_state["step"] > 0:
+        logger.info(
+            f"  Will skip the first {training_state['step']} steps. If this takes a lot of time,"
+            " you can add the `--ignore_data_skip` flag to your launch command, but you will resume the"
+            " training on data already seen by your model."
+        )
+        for step in tqdm(range(training_state["step"]), desc=f"Skipping data for {training_state['step']} steps...", position=1, leave=False):
+            try:
+                samples = next(train_loader)
```

Comment:
This is going to be quite slow right iterating over all the data that we're wanting to skip? Could be worth looking into `accelerate`'s function [`skip_first_batches`](https://github.com/huggingface/accelerate/blob/69e4c3c54da3201eda288b500d138761e7a5221c/src/accelerate/data_loader.py#L812)

Reply:
Then I think we would only need one iterator over `num_train_steps` (i.e. this loop and the training loop would collapse down into one)

Reply:
We really like the suggestion, but we're not sure it works when recovering after more than 1 epoch. Since this is streaming and we are training on number of steps, we could easily go over 1 epoch. The alternative would be to have a `steps_per_epoch` in the `training_state` so we can skip only `training_state["step"] % steps_per_epoch` batches. In any case, I think we'll experiment a bit in a different PR.